src/solvers/refinement/string_refinement.cpp

```diff
@@ -1003,41 +1006,54 @@ exprt string_refinementt::negation_of_not_contains_constraint(
     to_integer(to_constant_expr(ubu), ub_int);
     if(ub_int<=lb_int)
     {
-      debug() << "empty constraint with current model" << eom;
+      debug() << "(string_refinement::check_axioms) empty constraint with "
+              << "current model, adding false" << eom;
       return false_exprt();
     }
   }
 
   exprt lbe=axiom.exists_lower_bound();
   exprt ube=axiom.exists_upper_bound();
 
+  INVARIANT(
+    lbe.id()==ID_constant && ube.id()==ID_constant,
+    string_refinement_invariantt("please"));
```

Comment:
Whoops, I forgot to take that away, I wasn't 100% sure it would be true, I'll remove it.